stage5.3_fuller_application/pycasa/ui/image_folder_view.py

```diff
@@ -0,0 +1,130 @@
+# General imports
+
+# ETS imports
+import numpy as np
+import pandas as pd
+from traits.api import Bool, Button, Enum, Instance, List, observe
+from traitsui.api import HGroup, Item, Label, ListStrEditor, ModelView, \
+    Spring, View
+from traitsui.ui_editors.data_frame_editor import DataFrameEditor
+
+# Local imports
+from ..model.image_folder import FILENAME_COL, ImageFolder, NUM_FACE_COL
+
+
+DISPLAYED_COLUMNS = [FILENAME_COL, NUM_FACE_COL] + [
+    'ApertureValue', 'ExifVersion', 'Model', 'Make', 'LensModel', 'DateTime',
+    'ShutterSpeedValue', 'ExposureTime', 'XResolution', 'YResolution',
+    'Orientation', 'GPSInfo', 'DigitalZoomRatio', 'FocalLengthIn35mmFilm',
+    'ISOSpeedRatings', 'SceneType'
+]
+
+
+class ImageFolderView(ModelView):
+    """ ModelView for an image folder object.
+    """
+    model = Instance(ImageFolder)
+
+    scan = Button("Scan for faces...")
+
+    # Filters widgets
+    view_filter_controls = Bool
+
+    filtered_data = Instance(pd.DataFrame)
+
+    year_mask = Instance(pd.Series)
+
+    selected_years = List
+
+    all_years = List
+
+    selected_make = Enum(["All", "Canon", "Nikon", "Sony", "Apple", "samsung"])
+
+    make_mask = Instance(pd.Series)
+
+    def traits_view(self):
+        view = View(
+            Item("model.path", style="readonly", show_label=False),
+            HGroup(
+                Spring(),
+                Item("view_filter_controls"),
+            ),
+            HGroup(
+                Item("all_years", label="Years",
+                     editor=ListStrEditor(selected="selected_years",
+                                          multi_select=True)
+                     ),
+                Item("selected_make", label="Camera make"),
+                visible_when="view_filter_controls",
+            ),
+            Item("filtered_data",
+                 editor=DataFrameEditor(columns=DISPLAYED_COLUMNS,
+                                        update="data_updated"),
+                 show_label=False, visible_when="len(model.data) > 0"),
+            HGroup(
+                Spring(),
+                Label("No images found. No data to show"),
+                Spring(),
+                visible_when="len(model.data) == 0",
+            ),
+            HGroup(
+                Spring(),
+                Item("scan", show_label=False,
+                     enabled_when="len(model.data) > 0"),
+                Spring(),
+            ),
+            resizable=True, height=800
+        )
+        return view
+
+    # Listener methods --------------------------------------------------------
+
+    @observe("scan")
+    def scan_for_faces(self, event):
+        self.model.compute_num_faces()
```

Comment:
Does it make sense to only compute faces on the filtered data?

Reply:
It would be nice but I would like to keep the code as simple as possible.